### PR TITLE
Simplify unread/read digest sorting

### PR DESCRIPTION
**Pull request by dan-searle**
This reduces the code needed in `next-article` for the new ribbons.

### CODE REVIEW

File: myft-notification/fetch-digest-data.js [hunk 1]
```diff
@@ -16,12 +16,34 @@ const checkDigestDataExist = ({ data = {} } = {}) => {
 
 const flattenDigestSections = data => {
 	data.user.digest.articles = data.user.digest.concepts.reduce((flatArticles, concept) => {
-		return flatArticles.concat(concept.articles)
+		return flatArticles.concat(concept.articles);
```
Comment (benbarnett) [inline myft-notification/fetch-digest-data.js]: 🙈

File: myft-notification/fetch-digest-data.js [hunk 2]
```diff
@@ -16,12 +16,34 @@ const checkDigestDataExist = ({ data = {} } = {}) => {
...
 };
 
+const decorateWithReadStatus = data => {
+	const readArticles = data.user.articlesFromReadingHistory ? data.user.articlesFromReadingHistory.articles : [];
+
+	readArticles.forEach(readArticle => {
+		const readArticleInDigest = data.user.digest.articles.find(digestArticle => digestArticle.id === readArticle.id);
+
+		if (readArticleInDigest) {
+			readArticleInDigest.read = true;
```
Comment (benbarnett) [inline myft-notification/fetch-digest-data.js]: Although we’re only using this for digest experiments at the minute, I'm wondering if `read` is ambiguous with 'read' as an access permission? Perhaps `hasBeenRead`?

File: myft-notification/fetch-digest-data.js [hunk 3]
```diff
@@ -16,12 +16,34 @@ const checkDigestDataExist = ({ data = {} } = {}) => {
 
 const flattenDigestSections = data => {
```
Comment (benbarnett) [inline myft-notification/fetch-digest-data.js]: Not part of this PR but reading it together with the promise chain below sort of suggests this function does something else. What do you think about calling this `extractArticlesFromSections` or something?

File: myft-notification/fetch-digest-data.js (hunk 2 above)
Comment (dan-searle) [reply myft-notification/fetch-digest-data.js]: Yes, that seems better. 👍
